warehouse/jobs/runner.go

```diff
@@ -0,0 +1,373 @@
+		asyncWhJob.log.Infof("WH-Jobs: updating async jobs table query %s, retry no : %d", sqlStatement, queryretry)
+		_, err := asyncWhJob.dbHandle.Query(sqlStatement)
+		if err == nil {
+			pkgLogger.Errorf("query: %s successfully executed", sqlStatement)
+			return
+		}
+	}
+	pkgLogger.Errorf("query: %s failed with Error : %s", sqlStatement, err.Error())
+}
+
+//returns status and errMessage
+//Only succeeded, executing & waiting states should have empty errMessage
+//Rest of the states failed, aborted should send an error message conveying a message
+func (asyncWhJob *AsyncJobWhT) getStatusAsyncJob(payload *StartJobReqPayload) (status string, errMessage error) {
+	asyncWhJob.log.Info("WH-Jobs: Getting status for wh async jobs %v", payload)
+	//Need to check for count first and see if there are any rows matching the jobrunid and taskrunid. If none, then raise an error instead of showing complete
+	sqlStatement := fmt.Sprintf(`SELECT status,error FROM %s WHERE jobrunid='%s' AND taskrunid='%s'`, warehouseutils.WarehouseAsyncJobTable, payload.JobRunID, payload.TaskRunID)
+	asyncWhJob.log.Infof("Query is %s", sqlStatement)
+	rows, err := asyncWhJob.dbHandle.Query(sqlStatement)
+	if err != nil {
+		return WhJobFailed, err
+	}
+
+	for rows.Next() {
+		var status string
+		var errMessage sql.NullString
+		err = rows.Scan(&status, &errMessage)
+		if err != nil {
+			asyncWhJob.log.Errorf("WH-Jobs: Error scanning rows %s\n", err)
+			return WhJobError, err
+		}
+		if status == WhJobFailed {
+			asyncWhJob.log.Infof("[WH-Jobs] Async Job with jobrunid: %s, taskrunid: %s is failed", payload.JobRunID, payload.TaskRunID)
+			if !errMessage.Valid {
+				return WhJobAborted, errors.New("failed for unknown reasons")
+			}
+			return WhJobFailed, errors.New(errMessage.String)
+		}
+		if status == WhJobAborted {
+			asyncWhJob.log.Infof("[WH-Jobs] Async Job with jobrunid: %s, taskrunid: %s is aborted", payload.JobRunID, payload.TaskRunID)
+			if !errMessage.Valid {
+				return WhJobAborted, errors.New("aborted for unknown reasons")
+			}
+			return WhJobAborted, errors.New(errMessage.String)
+		}
+		if status != WhJobSucceeded {
+			asyncWhJob.log.Infof("[WH-Jobs] Async Job with jobrunid: %s, taskrunid: %s is under processing", payload.JobRunID, payload.TaskRunID)
+			return WhJobExecuting, err
+		}
+
+	}
+	asyncWhJob.log.Infof("[WH-Jobs] Async Job with jobrunid: %s, taskrunid: %s is complete", payload.JobRunID, payload.TaskRunID)
+	return WhJobSucceeded, err
+}
+
+//convert to pgNotifier Payload and return the array of payloads
+func getMessagePayloadsFromAsyncJobPayloads(asyncjobs []AsyncJobPayloadT) ([]pgnotifier.JobPayload, error) {
+	var messages []pgnotifier.JobPayload
+	for _, job := range asyncjobs {
+		message, err := json.Marshal(job)
+		if err != nil {
+			return messages, err
+		}
+		messages = append(messages, message)
+	}
+	return messages, nil
+}
+
+func getSinglePayloadFromBatchPayloadByTableName(asyncJobs *[]AsyncJobPayloadT, tableName string) int {
```

Comment:
Let's not use a pointer to array since arrays are already pointers. We don't need to deference them as well.